src/base.jl

```diff
@@ -1457,6 +1457,7 @@ end
 function handle_component_addition!(sys::System, component::RegulationDevice)
     copy_forecasts!(component.device, component)
     remove_component!(sys, component.device)
+    set_unit_system!(component.device, component.internal.units_info)
```

Comment:
I don't think this is necessary.  It is already handled by same call made in `add_component!`.